tests/h/subscribers_test.py

```diff
@@ -219,6 +222,39 @@ def test_it_enqueues_delete_annotation_celery_task_for_delete(
         delete_annotation.delay.assert_called_once_with(event.annotation_id)
         assert not add_annotation.delay.called
 
+    @pytest.mark.parametrize(
+        "action,method",
+        (
+            ("create", "add_annotation_by_id"),
+            ("update", "add_annotation_by_id"),
+            ("delete", "delete_annotation_by_id"),
+        ),
+    )
+    @pytest.mark.usefixtures("with_synchronous_flag")
+    def test_it_calls_sync_service(
+        self, action, pyramid_request, search_index, method, transaction_manager
+    ):
+        event = AnnotationEvent(pyramid_request, {"id": "any"}, action)
+
+        subscribers.sync_annotation(event)
+
+        transaction_manager.__enter__.assert_called_once()
+        getattr(search_index, method).assert_called_once_with(event.annotation_id)
+        transaction_manager.__exit__.assert_called_once()
+
+    @pytest.fixture(autouse=True)
+    def transaction_manager(self, pyramid_request):
+        pyramid_request.tm = mock.MagicMock(spec=["__enter__", "__exit__"])
+        return pyramid_request.tm
```

Comment:
```suggestion
    @pytest.fixture(autouse=True)
    def transaction_manager(self, pyramid_request):
        from transaction import TransactionManager
        pyramid_request.tm = mock.create_autospec(
            TransactionManager, instance=True, spec_set=True
        )
        return pyramid_request.tm
```

Let's autospec this just so that if `transaction`'s interface ever changes, the tests will pick up on that. (The `import` will have to go at the top of the file instead, of course.)